git/ssh_config_test.go

```diff
@@ -15,19 +20,97 @@ func eq(t *testing.T, got interface{}, expected interface{}) {
 	}
 }
 
-func Test_sshParse(t *testing.T) {
-	m := sshParse(strings.NewReader(`
-	Host foo bar
-		HostName example.com
-	`), strings.NewReader(`
-	Host bar baz
-	hostname %%%h.net%%
-	`))
+func createTempFile(t *testing.T, prefix string) *os.File {
+	t.Helper()
+
+	dir, err := homedir.Dir()
+	if err != nil {
+		t.Errorf("Could not find homedir: %s", err)
+	}
+
+	tempFile, err := ioutil.TempFile(filepath.Join(dir, ".ssh"), prefix)
+	if err != nil {
+		t.Errorf("Could create a temp file: %s", err)
+	}
+
+	t.Cleanup(func() {
+		tempFile.Close()
+		os.Remove(tempFile.Name())
+	})
+
+	return tempFile
+}
+
+func Test_parse(t *testing.T) {
+	includedTempFile := createTempFile(t, "included")
+	includedConfigFile := `
+Host webapp
+	HostName webapp.example.com
+	`
+	fmt.Fprint(includedTempFile, includedConfigFile)
```

Comment:
I am not very happy with this test, but I need to created a temp file under `~/.ssh/` dir in order to test inclusion logic.. Any suggestion to improve this will be welcome 😃